src/tasks/UploadSlice.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import {
+  ErrorMappings,
+  Headers,
+  HttpMethod,
+  Parsable,
+  ParsableFactory,
+  RequestAdapter,
+  RequestInformation,
+} from "@microsoft/kiota-abstractions";
+import { HeadersInspectionOptions } from "@microsoft/kiota-http-fetchlibrary";
+import { UploadResult, UploadSession } from "./LargeFileUploadTask";
+
+const binaryContentType = "application/octet-stream";
+
+/**
+ * @class
+ * Class for UploadSlice
+ */
+export class UploadSlice<T extends Parsable> {
+  constructor(
+    readonly requestAdapter: RequestAdapter,
+    readonly sessionUrl: string,
+    readonly rangeBegin: number,
+    readonly rangeEnd: number,
+    readonly totalSessionLength: number,
+    readonly parsableFactory: ParsableFactory<T>,
+    readonly errorMappings: ErrorMappings,
+  ) {}
+
+  /**
+   * Uploads a slice of the file to the server.
+   *
+   * @param {ReadableStream<Uint8Array>} stream - The stream of the file slice to be uploaded.
+   * @returns {Promise<UploadResult<T> | undefined>} - The result of the upload operation.
+   */
+  public async uploadSlice(stream: ReadableStream<Uint8Array>): Promise<UploadResult<T> | undefined> {
+    const data = await this.readSection(stream, this.rangeBegin, this.rangeEnd);
+    const requestInformation = new RequestInformation(HttpMethod.PUT, this.sessionUrl);
+    requestInformation.headers = new Headers([
+      ["Content-Range", new Set([`bytes ${this.rangeBegin}-${this.rangeEnd - 1}/${this.totalSessionLength}`])],
+      ["Content-Length", new Set([`${this.rangeEnd - this.rangeBegin}`])],
+    ]);
+    requestInformation.setStreamContent(data, binaryContentType);
+
+    const headerOptions = new HeadersInspectionOptions({ inspectResponseHeaders: true });
+    requestInformation.addRequestOptions([headerOptions]);
+
+    let itemResponse = await this.requestAdapter.send<T>(requestInformation, this.parsableFactory, this.errorMappings);
+
+    const locations = headerOptions.getResponseHeaders().get("location");
+
+    let uploadSession: UploadSession | null = null;
+    if (itemResponse) {
```

Comment:
Out of curiosity, when uploading to the onedrive, won't the `parsableFactory` be the factory of a `driveItem`? 
If so, we will always attempt to deserialize the reponse into a driveItem. This means that properties here (e.g expirationDateTime) will be in the additionalData right?

Reply:
checking both additional data and the main body